R/http.R

```diff
@@ -73,6 +76,65 @@ httpRequestWithBody <- function(service,
     contentFile = file,
     certificate = certificate
   )
+  handleResponse(httpResponse, error_call = error_call)
+}
+
+handleResponse <- function(response, error_call = caller_env()) {
+  reportError <- function(msg) {
+    req <- response$req
+    url <- paste0(req$protocol, "://", req$host, req$port, req$path)
+
+    cli::cli_abort(
+      c("<{url}> failed with HTTP status {response$status}", msg),
+      class = c(paste0("rsconnect_http_", response$status), "rsconnect_http"),
+      call = error_call
+    )
+  }
+
+  if (isContentType(response, "application/json")) {
+    # parse json responses
+    if (nzchar(response$content)) {
+      json <- jsonlite::fromJSON(response$content, simplifyVector = FALSE)
+    } else {
+      json <- list()
+    }
+
+    if (response$status %in% 200:399) {
+      out <- json
+    } else if (!is.null(json$error)) {
+      reportError(json$error)
+    } else {
+      reportError(paste("Unexpected json response:", response$content))
```

Comment:
Note to self: `readHttpResponse` ensures that `response$content` is a character vector, not a connection, so it is OK to use its value repeatedly.

Given that we have already parsed the content, we could pretty-print the object, but that might introduce representation adjustments (because R). OK using the raw `content` here.